### ISSUE TITLE
Why the CLISP support was dropped?

### ISSUE THREAD

**Issue by Sasanidas** [stumpwm/stumpwm#929]
I saw that from version 1.0 onwards https://github.com/stumpwm/stumpwm/issues/327#issuecomment-282495665 , other implementation support (including CLISP) was dropped.

Is there any good reason for this? I mean, I love SBCL but I also like to try other implementations time to time, and I found CLISP to be pretty interesting, and even tho the "release" cicle is slow, it's [still in development](https://gitlab.com/gnu-clisp/clisp).

Regards.

**Comment by jorams** [comment]
CLISP required a bunch of workarounds and special fixes due to non-compliance and incompatibilities, and they haven't published a release in over 11 years. The fact that there's ongoing development doesn't help since distributions don't ship random commits, they ship releases.

**Comment by Sasanidas** [comment]
> CLISP required a bunch of workarounds and special fixes due to non-compliance and incompatibilities,

When you say "non-compliance" do you mean that CLISP is not fully ANSI compliance? (even with the -ansi cli option)

> they haven't published a release in over 11 years. The fact that there's ongoing development doesn't help since distributions don't > ship random commits, they ship releases.

Fair enough, but is the are any possibility to make it compatible for the current releases? (even in a separate branch)

 I'm willing to invest some time if the StumpWM team see any value in it.

Regards.

**Comment by jorams** [comment]
> When you say "non-compliance" do you mean that CLISP is not fully ANSI compliance? (even with the -ansi cli option)

Indeed. See 53bc4eab3c559b624c5832e8d17ee59c264d259e for a random example.

> Fair enough, but is the are any possibility to make it compatible for the current releases? (even in a separate branch)
> 
> I'm willing to invest some time if the StumpWM team see any value in it.

It's possible, not sure how much work it would be. We dropped support because maintenance wasn't worth the effort.

**Comment by jorams** [comment]
> But in this case, it's stumpwm that wasn't conforming !!!

Good point, that's not a great example and I can't quickly find an example of non-compliance. (I just searched for "clisp" and this commit seemed relevant.) However, the associated issue #287 is a pretty good example of CLISP support not being worth it. It completely breaks StumpWM, but it was reported by me (a non-user of CLISP) and it didn't receive any input for months until I fixed it after a painful debugging session.

> So you mean that writing conforming code is not worth the effort and you prefer to use implementation specific code?
> 
> I don't see how it's an effort, it's just a different language ! Try to target Common Lisp, instead of SBCL !

Writing portable code is good, but StumpWM does a bunch of stuff that simply can't be done in portable CL. That's where the effort comes in. Anything touching such issues will have to be implemented for every supported implementation, even if nobody uses them. 3d21db8b6ce23502d755b3316731dd5b3b6fb490 is a commit that removed such code.
